packages/astroport/src/generator.rs

```diff
@@ -9,6 +9,10 @@ use serde::{Deserialize, Serialize};
 pub struct InstantiateMsg {
     /// Address that can change contract settings
     pub owner: String,
+    /// Address of factory contract
+    pub factory: String,
+    /// Address that can set active generators and their alloc points
+    pub generator_controller: String,
```

Comment:
Besides that generator contract is already deployed I suggest to make it `Option<String>` because Generator controller is also needs generator address.  It is vicious circle.

Reply:
Ok, it makes sense. Changed, thanks